packages/eas-cli/src/commands/project/onboarding.ts

```diff
@@ -0,0 +1,224 @@
+    await installDependenciesAsync({
+      projectDir: targetProjectDir,
+    });
+    await vcsClient.trackFileAsync('package-lock.json');
+
+    const shouldSetupCredentials =
+      actor.preferences.onboarding.deviceType === OnboardingDeviceType.Device &&
+      actor.preferences.onboarding.environment === OnboardingEnvironment.DevBuild;
+    if (shouldSetupCredentials) {
+      Log.log('🔑 Now we need to set up build credentials for your project:');
+      await new SetUpBuildCredentialsCommandAction(
+        actor,
+        graphqlClient,
+        vcsClient,
+        analytics,
+        await getPrivateExpoConfigWithProjectIdAsync({
+          projectDir: targetProjectDir,
+          graphqlClient,
+          actor,
+        }),
+        getDynamicPrivateProjectConfigGetter({
+          projectDir: targetProjectDir,
+          graphqlClient,
+          actor,
+        }),
+        platform,
+        'development',
+        targetProjectDir
+      ).runAsync();
+    }
+
+    if (await vcsClient.hasUncommittedChangesAsync()) {
+      Log.log(
+        '📦 We will now commit the changes made by the configuration process and push them to GitHub:'
+      );
+      Log.log();
+      Log.log('🔍 Checking for changes in the repository...');
+      await vcsClient.showChangedFilesAsync();
+      await reviewAndCommitChangesAsync(
+        vcsClient,
+        `[EAS onboarding] Install dependencies${
+          shouldSetupCredentials ? 'and set up build credentials' : ''
+        }`,
+        { nonInteractive: false }
+      );
+      Log.log('📤 Pushing changes to GitHub...');
+      await runGitPushAsync({
+        targetProjectDir,
+      });
+    }
+
+    Log.log();
+    Log.log('🎉 We finished configuring your project.');
+    Log.log('🚀 You can now go back to the website to continue.');
+  }
+}
+
+// we can't get this automated by using command context because when we run a command the project directory doesn't exist yet
+async function getPrivateExpoConfigWithProjectIdAsync({
+  projectDir,
+  graphqlClient,
+  actor,
+}: {
+  projectDir: string;
+  graphqlClient: ExpoGraphqlClient;
+  actor: Actor;
+}): Promise<CredentialsContextProjectInfo> {
+  const expBefore = getPrivateExpoConfig(projectDir);
+  const projectId = await validateOrSetProjectIdAsync({
+    exp: expBefore,
+    graphqlClient,
+    actor,
+    options: {
+      nonInteractive: false,
+    },
+  });
+  const exp = getPrivateExpoConfig(projectDir);
+  return {
+    exp,
+    projectId,
+  };
+}
+
+// we can't get this automated by using command context because when we run a command the project directory doesn't exist yet
+function getDynamicPrivateProjectConfigGetter({
+  projectDir,
+  graphqlClient,
+  actor,
+}: {
+  projectDir: string;
+  graphqlClient: ExpoGraphqlClient;
+  actor: Actor;
+}): DynamicConfigContextFn {
+  return async (options?: ExpoConfigOptions) => {
+    const expBefore = getPrivateExpoConfig(projectDir, options);
```

Comment:
Why not reuse `getPrivateExpoConfigWithProjectIdAsync` here?

Reply:
oh, sure